### PR TITLE
Initial EDOMH integration

### LINKED ISSUE #703: EDOMH Integration
Request for integration with the odyssey materials helper
https://github.com/jixxed/ed-odyssey-materials-helper

recently implemented in EDSY: https://github.com/taleden/EDSY/commit/2dd2981873f9859a6d8c0ed04b515517e34c554e

works by creating a url that contains compressed json base64 encoded in the format:
edomh://coriolis/?<compressed json base64 encoded>

json follows the format:
{
    "version":1,
    "name":"ship build name",
    "items": [
        {
            "item": "Hpt_BeamLaser_Fixed_Small",
            "blueprint": "Weapon_Efficient",
            "grade": 5,
            "highestGradePercentage":0.8
        },
        {
            "item": "Hpt_BeamLaser_Fixed_Small",
            "blueprint": "special_weapon_damage"
        }
    ]
}

experimental effect do not require the grade and highestGradePercentage fields.
only blueprints and experimental effects need to included in the json, no modules. 

the url can then be embedded  on the site and contains all the information to construct a wishlist in the app.

### PR DESCRIPTION
**Pull request by Sid127**
Adds a button below Export to EDEngineer that exports to EDOMH. A bit hacky on the code, but works flawlessly across testing with multiple builds.

I am yet to figure out how to have the ships build name imported as well, but I figured it'd be good to push this and see if the community can contribute for the same with a PR

Closes #703

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -124,6 +124,7 @@
   "dependencies": {
     "@babel/polyfill": "^7.0.0",
     "auto-bind": "^5.0.1",
+    "base64url": "^3.0.1",
```
Comment (felixlinker) [inline package.json:127]: Is this dependency really necessary? Could we also go for a combination of `atob` and `encodeURIComponent` (and the respective inverses)?

File: src/app/components/ModalShoppingList.jsx [hunk 2]
```diff
@@ -146,6 +147,64 @@ export default class ModalShoppingList extends TranslatedComponent {
...
+    let JSONString = JSON.stringify(baseJson)
+    let deflated = zlib.deflateSync(JSONString)
+    
+    //actually encode
+    let link = base64url.encode(deflated)
+    link = "edomh://coriolis/?" + link;
+
+    window.open(link, "_self")
+  }
+
```
Comment (felixlinker) [inline src/app/components/ModalShoppingList.jsx:207]: Code looks good! This could looks like it might also work for edsy. This is just SLEF, right?

**Review by felixlinker — COMMENTED** [review]
Code looks good and not hacky at all. Just one comment on the dependency, and one thought about edsy.

File: src/app/components/ModalShoppingList.jsx (hunk 2 above)
Comment (Sid127) [reply src/app/components/ModalShoppingList.jsx:207]: It isn't SLEF, EDOMH uses its own format :\

Comment (jixxed) [reply src/app/components/ModalShoppingList.jsx:207]: SLEF wouldn't fit inside the url :)

**Pull request merged**